### PR TITLE
Move ImageNet-R, ImageNet-C, and other example tasks

### LINKED ISSUE #84: [imagenet_r.py] wrong md5
Running the imagenet_r.py example causes the following error:
```
Downloading https://people.eecs.berkeley.edu/~hendrycks/imagenet-r.tar to ../../../../test_data\imagenet_r\imagenet-r.tar
2191080448it [07:01, 5195316.09it/s]                                
Traceback (most recent call last):
  File "E:\git_projects\bethgelab\icml-2022\shifthappens\tasks\imagenet_r\imagenet_r.py", line 81, in <module>
    sh_benchmark.evaluate_model(ResNet18(device="cuda", max_batch_size=128), "../../../../test_data")
  File "E:\git_projects\bethgelab\icml-2022\shifthappens\benchmark.py", line 150, in evaluate_model
    task.setup()
  File "E:\git_projects\bethgelab\icml-2022\shifthappens\tasks\imagenet_r\imagenet_r.py", line 40, in setup
    sh_utils.download_and_extract_archive(
  File "E:\git_projects\bethgelab\icml-2022\shifthappens\utils.py", line 127, in download_and_extract_archive
    tv_utils.download_url(url, data_folder, filename, md5)
  File "C:\Python310\lib\site-packages\torchvision\datasets\utils.py", line 152, in download_url
    raise RuntimeError("File not found or corrupted.")
RuntimeError: File not found or corrupted.
```
I think there is a problem with the md5 checksum since if you open the archive without it (just passing `None` instead of `A61312130A589D0CA1A8FCA1F2BD3337`), it does not cause any error.

Should we change m5? Or did we copy this wrong from some repo?

### PR DESCRIPTION
**Pull request by kotekjedi**
* examples/ -> tasks/
* raccoons_OOD.py -> raccoons_ood.py

TODO:
- [x] Fix #84

### CODE REVIEW

**Comment by kotekjedi** [comment]
I think that [#75](https://github.com/shift-happens-benchmark/icml-2022/pull/75) should be merged first, and then we can update paths here as well.

File: shifthappens/tasks/imagenet_c/README.rst [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (stes) [inline shifthappens/tasks/imagenet_c/README.rst:1]:
```suggestion
```

File: shifthappens/tasks/imagenet_r/imagenet_r.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def setup(self):
             # download data
             for file_name, url, md5 in self.resources:
                 sh_utils.download_and_extract_archive(
-                    url, self.data_root, md5, file_name
+                    url, self.data_root, None, file_name
```
Comment (stes) [inline shifthappens/tasks/imagenet_r/imagenet_r.py:41]: Could you add the hash? If you dont want to recompute, they are probably also published somewhere.

File: shifthappens/tasks/raccoons_ood/README.rst [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (stes) [inline shifthappens/tasks/raccoons_ood/README.rst:1]:
```suggestion
```

File: shifthappens/tasks/raccoons_ood/README.rst [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+Example for a Shift Happens task on Raccoons dataset
+====================================================
+This task aims to evaluate models' out-of-distribution (OOD) detection on 200 raccoon images.
+Raccoons are not presented in ImageNet classes, so the task uses models'
+confidences (maximal predicted class probability) for the ImageNet validation set and
+raccoons images (ImageNet samples treated as class 1 and raccoons as class 0) to measure
+AUROC and FPR at TPR equal 0.95.
+
+The original dataset was collected by Dat Tran for the object detection task
+and can be found at https://github.com/datitran/raccoon_dataset.
```
Comment (stes) [inline shifthappens/tasks/raccoons_ood/README.rst:11]: @j-cb do you want to extend the readme for this task? (just propose as a change to this PR).

**Review by stes — CHANGES_REQUESTED** [review]
Thanks @kotekjedi .

According to the contribution guide you can move the contents of imagenet_c.py directly into the `__init__.py`, or at least make sure in the `__init__.py` that the task is registered. (maybe the latter option is better)

**Comment by zimmerrol** [comment]
@kotekjedi Once you have fixed the merge conflcits and fixed the hash, we can merge this PR.

**Comment by kotekjedi** [comment]
Done. I updated the cash (for me, it was an upper-case problem).

**Comment by zimmerrol** [comment]
@kotekjedi Can you please update the READMEs for ImageNet-C/R based in the current docstring of the classes & modules (e.g., the references to the papers must be included).

**Review by zimmerrol — APPROVED** [review]
LGTM

**Pull request merged**